libcaf_core/caf/monitor.test.cpp

```diff
@@ -0,0 +1,131 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/event_based_actor.hpp"
+
+using namespace caf;
+
+namespace {
+
+WITH_FIXTURE(test::fixture::deterministic) {
+
+TEST("monitoring another actor") {
+  auto client_spawn = [](event_based_actor*) {
+    return behavior{
+      [](message) {},
+    };
+  };
+  auto client1 = sys.spawn(client_spawn);
+  auto client2 = sys.spawn(client_spawn);
+  auto client3 = sys.spawn(client_spawn);
+  SECTION("monitoring an actor invokes the down handler") {
+    auto call_count = std::make_shared<int32_t>(0);
+    auto observer = sys.spawn([=](event_based_actor* self) {
+      self->monitor(client1);
+      self->monitor(client2);
+      self->monitor(client3);
+      self->set_down_handler(
+        [call_count](const down_msg&) { *call_count += 1; });
+      return behavior{
+        [](int32_t) {},
+      };
+    });
+    inject_exit(client1);
+    expect<down_msg>().with(std::ignore).from(client1).to(observer);
+    check_eq(*call_count, 1);
+    inject_exit(client2);
+    expect<down_msg>().with(std::ignore).from(client2).to(observer);
+    check_eq(*call_count, 2);
+    inject_exit(client3);
+    expect<down_msg>().with(std::ignore).from(client3).to(observer);
+    check_eq(*call_count, 3);
+  }
+  SECTION("monitoring with special lambda invokes the lambda") {
```

Comment:
What is a "special" lambda supposed to be?

```suggestion
  SECTION("monitoring with a callback") {
```